app/initializers/fastboot/ajax.js

```diff
@@ -1,6 +1,19 @@
 /* globals najax */
+import Ember from 'ember';
+
+const { get } = Ember;
 
 var nodeAjax = function(options) {
+  let httpRegex = /^https?:\/\//
+
+  if (!httpRegex.test(options.url)) {
+    try {
+      options.url = get(this, 'fastboot.host') + options.url
+    } catch (fbError) {
+      throw new Error('You are using Ember Data with no host defined in your adapter. This will attempt to use the host of the FastBoot request, which is not configured for the current host of this request. Please set the hostWhitelist property for in your environement.js. FastBoot Error: ' + fbError.message);
```

Comment:
Right at the very end of this long error message `environment` is mis-spelled :-)